tasks/base_tasks.py

```diff
@@ -1592,3 +1610,133 @@ def collect_meta_wrappers(test_module=None, test_all=True):
             yield t, params
             if not test_all:
                 break
+
+
+class CreateRepoTable(Task):
+    schema = 'repo'
+    table = 'filerepo'
+
+    def run(self):
+        session = current_session()
+
+        create_schema = '''
+                        CREATE SCHEMA IF NOT EXISTS "{schema}"
+                        '''.format(schema=self.output().schema)
+        session.execute(create_schema)
+
+        create_table = '''
+                        CREATE TABLE IF NOT EXISTS "{schema}".{table} (
+                            id TEXT,
+                            version NUMERIC,
+                            checksum TEXT,
+                            url TEXT,
+                            path TEXT,
+                            added TIMESTAMP,
+                            PRIMARY KEY(id, version)
+                        );
+                        '''.format(schema=self.output().schema,
+                                   table=self.output().tablename)
+        session.execute(create_table)
+        session.commit()
+
+    def output(self):
+        return PostgresTarget(schema=self.schema, tablename=self.table, non_empty=False)
+
+
+class RemoteDownloader:
+    def download(self, url, output_path):
+        raise NotImplementedError
+
+
+class BaseDownloader(RemoteDownloader):
+    def download(self, url, output_path):
+        urllib.request.urlretrieve(url, output_path)
```

Comment:
Neither `BaseDownloader` nor `RemoteDownloader` rely on object state, and using them for inheritance leads to checks such as `if not issubclass(self.downloader.__class__, RemoteDownloader):`. What if those classes were removed and this `download` method became a normal function?

Reply:
You are right... part of my heart still runs on Java. Changing it :)